File: 이재진/롤케이크자르기.py
def solution(topping):
    answer = 0
    a = dict()
    b = dict()
    for i in topping:
        if i in a:
            a[i] += 1
        else:
            a[i] = 1
    for i in topping:
        if len(a.keys()) == len(b.keys()):
            answer += 1
        if a[i] == 1:
            del a[i]
        else:
            a[i] -= 1
        
        if i in b:
            b[i] += 1
        else:
            b[i] = 1

    return answer

File: 정종길/프로그래머스/롤케이크 자르기.java
import java.util.*;

class Solution {
    public int solution(int[] topping) {
        int answer = 0;
        HashMap<Integer, Integer> older = new HashMap<>();
        HashMap<Integer, Integer> younger = new HashMap<>();
        
        
        for(int top :topping){
            int check = older.getOrDefault(top, 0);
            older.put(top, check + 1);
        }
        
         for(int top :topping){
            int yCheck = younger.getOrDefault(top, 0);
            younger.put(top, yCheck + 1);
            
            Integer oCheck = older.get(top);
             
            if(oCheck == null || oCheck == 1){
                older.remove(top);
            } else {
                older.put(top, oCheck - 1);
            }
            
            if(older.size() == younger.size()){
                answer += 1;
            }
        }

        return answer;
    }
}